docs/developer.md

```diff
@@ -0,0 +1,64 @@
+# Mobile App Developer's Guide
+
+This guide outlines the best practices to follow when you’re building a mobile app for the Province of B.C. 
+
+
```

Comment:
we need a ToC or some sort of paragraph to set up what the reader is about to experience. are these are things one needs to consider? is this more about how to use the guide?

all the sections below are H2s and don't give any visual emphasis on what the reader needs to focus on